### PR TITLE
fix: Don't follow circular symlinks

### PR DESCRIPTION
**Pull request by joshhunt**
Fixes the `Error: ELOOP: too many symbolic links encountered` exception that happens during the walkdir when the working directory contains circular symlinks.

I think this is the right logic to check this, and added some tests to try and catch the common scenarios.

### CODE REVIEW

**Comment by phated** [comment]
@joshhunt could we just check if the error is ELoop and bail?

**Comment by joshhunt** [comment]
@phated I think this is in a good spot - open for feedback, other approaches, or scenarios to test!

File: test/index.js [hunk 1]
```diff
@@ -765,12 +765,48 @@ function suite(moduleName) {
...
+        },
+
+        // And it should follow a symlink to a parent directory (circular symlink) without blowing up
+        {
+          cwd: dir,
+          base: dir + '/fixtures/symlinks',
+          path:
+            dir +
+            '/fixtures/symlinks/symlink-dest/hey/isaidhey/whatsgoingon/test.txt',
+        },
```
Comment (phated) [inline test/index.js:799]: I need to check if this is a duplicate glob getting through because we wouldn't want to emit it twice.

**Review by phated — APPROVED** [review]
Thanks @joshhunt! I'm going to merge this and then do some cleanup to integrate it with my changes to avoid blowing the stack.

**Pull request merged**